### PR TITLE
Scanner view

### PR DESCRIPTION
**Pull request by carcos-wolox**
### Summary
**Trello card:** https://trello.com/c/RQi2RhbG/2-scaner-de-c%C3%B3digos
Code scanner view

### Screencast
https://drive.google.com/open?id=1U2S7f2E3G51XT6FcHXVUOmTGVZUXbPwd

### CODE REVIEW

File: Cookbook-ios/CodeScanner/CodeScannerController.swift [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+                                                  codeOutputHandler: handleCode)
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        feature = Features.codeScanner
+        configureNavigationBar()
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(true)
```
Comment (gmazzei) [inline Cookbook-ios/CodeScanner/CodeScannerController.swift]: Is `true` necessary?

File: Cookbook-ios/CodeScanner/CodeScannerController.swift [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+private extension CodeScannerController {
+    func validateCameraAccess() {
+        guard UIImagePickerController.isSourceTypeAvailable(.camera) else {
+            showErrorAlert(message: "You don't have a camera")
+            return
+        }
+        
+        AVCaptureDevice.requestAccess(for: AVMediaType.video) { [unowned self] hasAccess in
+            guard hasAccess else {
+                self.showErrorAlert(message: "Your camera access was rejected. Plase change this configuration in Settings")
```
Comment (gmazzei) [inline Cookbook-ios/CodeScanner/CodeScannerController.swift]: Add the "e" in "Please" 😉

File: Cookbook-ios/Utils/Features.swift [hunk 3]
```diff
@@ -55,6 +55,27 @@ enum Features: CaseIterable {
...
+                - dataMatrix
+                - ean13
+                - ean8
+                - interleaved2of5
+                - itf14
+                - pdf417
+                - qr
+                - upce
+            
+            - We validate if the user has camera and if he/she has accepted the access permision.
```
Comment (gmazzei) [inline Cookbook-ios/Utils/Features.swift]: Add another "s" in "permission"

**Review by gmazzei — APPROVED** [review]
Looks good! 🏆 
I just left some comments above.

**Review by danielcharry — APPROVED** [review]

**Pull request merged**